### PR TITLE
Update: Report unexpected tagged template (fixes #4210)

### LINKED ISSUE #4210: no-unexpected-multiline should warn for tagged template strings [$15]
`no-unexpected-multiline` should warn for this case:

```js
let x = function() {}
`hello`
```

To prevent it from being parsed like this:

```js
let x = function() {}`hello`
```

For the full previous discussion about this topic, see: https://github.com/eslint/eslint/issues/4169#issuecomment-149064446

<bountysource-plugin>
---
There is a **[$15 open bounty](https://www.bountysource.com/issues/27562926-no-unexpected-multiline-should-warn-for-tagged-template-strings?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by rpatil26**
Checks and reports function expression being treated as tagged template accidently.

### CODE REVIEW

**Comment by michaelficarra** [comment]
Haha #4210 was not asking for this rule to look for function expressions. That was just an example. This was supposed to catch cases where the tag and the rest of the template are on different lines, just like how the rest of the rule works.

**Comment by rpatil26** [comment]
:smiley:  I missed the point then. Let me fix it

**Comment by rpatil26** [comment]
@michaelficarra I have updated it and also added a couple of more test cases. Can you please have a look once more?

File: lib/rules/no-unexpected-multiline.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ module.exports = function(context) {
 
     var FUNCTION_MESSAGE = "Unexpected newline between function and ( of function call.";
     var PROPERTY_MESSAGE = "Unexpected newline between object and [ of property access.";
+    var TAGGED_TEMPLATE_MESSAGE = "Unexpected newline between tag and the template string.";
```
Comment (michaelficarra) [inline lib/rules/no-unexpected-multiline.js]: I would go with

> Unexpected newline between template tag and template literal.

**Comment by michaelficarra** [comment]
LGTM.

**Comment by rpatil26** [comment]
Done. Thanks @michaelficarra

**Pull request merged**